test_conformance/api/test_queries.cpp

```diff
@@ -476,30 +476,124 @@ REGISTER_TEST(get_command_queue_info_compatibility)
+        clCreateContext(NULL, 1, &device, nullptr, nullptr, &error);
+    test_error_ret(error, "Unable to create testing context", CL_SUCCESS);
+
+    cl_uint num_devices = 0;
+    error = clGetContextInfo(context_to_test, CL_CONTEXT_NUM_DEVICES,
+                             sizeof(cl_uint), &num_devices, nullptr);
+    test_error(error, "clGetContextInfo CL_CONTEXT_NUM_DEVICES failed\n");
+
+    test_assert_error(num_devices == 1,
+                      "Context must contain exactly one device\n");
+
+    cl_device_id comp_device = nullptr;
+    error = clGetContextInfo(context_to_test, CL_CONTEXT_DEVICES,
+                             sizeof(cl_device_id), &comp_device, nullptr);
+    test_error(error, "clGetContextInfo CL_CONTEXT_DEVICES failed\n");
+
+    test_assert_error(device == comp_device,
+                      "Unexpected result returned by CL_CONTEXT_DEVICES query");
+
+    // query context properties against valid size
     size_t size;
     cl_context_properties props;
+    error = clGetContextInfo(context, CL_CONTEXT_PROPERTIES, sizeof(props),
+                             &props, &size);
+    test_error(error, "Unable to get context props");
 
-    error = clGetContextInfo( context, CL_CONTEXT_PROPERTIES, sizeof( props ), &props, &size );
-    test_error( error, "Unable to get context props" );
-
-    if (size == 0) {
+    if (size == 0)
+    {
         // Valid size
         return 0;
-    } else if (size == sizeof(cl_context_properties)) {
+    }
+    else if (size == sizeof(cl_context_properties))
+    {
         // Data must be NULL
-        if (props != 0) {
+        if (props != 0)
+        {
             log_error("ERROR: Returned properties is no NULL.\n");
             return -1;
         }
         // Valid data and size
         return 0;
     }
     // Size was not 0 or 1
-    log_error( "ERROR: Returned size of context props is not valid! (expected 0 or %d, got %d)\n",
-              (int)sizeof(cl_context_properties), (int)size );
+    log_error("ERROR: Returned size of context props is not valid! (expected 0 "
+              "or %d, got %d)\n",
+              (int)sizeof(cl_context_properties), (int)size);
     return -1;
 }
 
+REGISTER_TEST(get_context_info_mult_devices)
+{
+    // query multi-device context and perform objects comparability test
+    cl_uint maxComputeUnits = 0;
+    cl_int err =
+        clGetDeviceInfo(device, CL_DEVICE_MAX_COMPUTE_UNITS,
+                        sizeof(maxComputeUnits), &maxComputeUnits, NULL);
+    test_error_fail(err, "Unable to get maximal number of compute units");
+
+    cl_device_partition_property partitionProp[] = {
+        CL_DEVICE_PARTITION_EQUALLY,
+        static_cast<cl_device_partition_property>(maxComputeUnits / 2), 0
+    };
+
+    cl_uint num_devices = 0;
+    // how many sub-devices can we create?
+    err = clCreateSubDevices(device, partitionProp, 0, nullptr, &num_devices);
+    if (err != CL_SUCCESS || num_devices < 2)
```

Comment:
Possible areas for improvement here:

1. Could we make the right queries to determine if the device is partitionable vs. trying to partition the device and handling any generated errors?  This isn't a negative test, so it would be nice not to rely on error behavior.
2. Could we handle other partitioning types so the test may run on more devices?

Reply:
Done as suggested, sadly I don't have partitionable device - I will appreciate a proper test, thanks.